core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -566,6 +574,13 @@ private[spark] class ExecutorAllocationManager(
     if (executorIds.contains(executorId)) {
       executorIds.remove(executorId)
       removeTimes.remove(executorId)
+      if (!removedExecutorIds.offer(executorId)) {
+        logWarning(s"The queue is full for tracking removed executors, will trim old " +
+          s"executor ids and offer again")
+        val trimSize = math.max(1, removedExecutorIds.size() / 10)
```

Comment:
It seems `trimSize` would always be `MAXIMUM_NUM_REMOVED_EXECUTORS` / 10.

Reply:
Yes.. we can optimize it.